### ISSUE TITLE
Documentation for request encoders points to wrong interfaces

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6028]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)| @therve reported|
|-|-|
|Trac ID|trac#6028|
|Type|enhancement|
|Created|2012-09-29 13:44:07Z|
|Branch|https://github.com/twisted/twisted/tree/fix-encoder-doc-6028|

When merging [#11528](https://github.com/twisted/twisted/issues/11528) I kept wrong links to interfaces when renaming them, it should be fxied. OTOH I'm not sure pointing to private interfaces is really useful.

<details><summary>Searchable metadata</summary>

```
trac-id__6028 6028
type__enhancement enhancement
reporter__therve therve
priority__normal normal
milestone__ 
branch__branches_fix_encoder_doc_6028 branches/fix-encoder-doc-6028
branch_author__therve therve
status__closed closed
resolution__wontfix wontfix
component__web web
keywords__documentation_easy documentation easy
time__1348926247000000 1348926247000000
changetime__1348994319000000 1348994319000000
version__None None
owner__therve therve
cc__jknight
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_2"></a>|@therve commented|
|-|-|

(In [35843]) Branching to 'fix-encoder-doc-6028'

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_3"></a>|@therve commented|
|-|-|

i removed the references as it seems more clear to me.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set owner to @therve|
|-|-|

Fix seems correct as far as it goes.

However I have a concern about [#11528](https://github.com/twisted/twisted/issues/11528), related to the SSL/zlib CRIME attack.  I wonder if the `encoders` argument to `Site` is safe at all (if it isn't, then maybe it's not worth applying documentation fixes).

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun commented|
|-|-|

therve offered to introduce a more focusable API for encoders before the next Twisted release, or revert the original merge to avoid committing us to supporting it.  That seems fine to me.  So, go ahead and merge this if you like (and please file a ticket for that adjusted API).  Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_6"></a>|@therve set status to `closed`|
|-|-|

I'll fix the documentation in the new ticket, [#6031](https://github.com/twisted/twisted/issues/6031).
